signature/mechanism_gpgme.go

```diff
@@ -0,0 +1,175 @@
+// +build !containers_image_openpgp
+
+package signature
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/mtrmac/gpgme"
+)
+
+// A GPG/OpenPGP signing mechanism, implemented using gpgme.
+type gpgmeSigningMechanism struct {
+	ctx          *gpgme.Context
+	ephemeralDir string // If not "", a directory to be removed on Close()
+}
+
```

Comment:
Why? `func newGPGSigningMechanismInDirectory(…) (SigningMechanism, …)` is sufficient to ensure that `gpgmeSigningMechanism` implements the interface. Or would that line serve another purpose?